src/iteration_summary.jl

```diff
@@ -1,12 +1,40 @@
-struct IterationSummary{P, M, C, V, E}
+struct IterationSummary{P, M, C, V, E, O}
     parameters :: P     # constrained
     ensemble_mean :: M  # constrained
     ensemble_cov :: C   # constrained
     ensemble_var :: V
     mean_square_errors :: E
+    objective_values :: O
     iteration :: Int
 end
 
+"""
+    eki_objective(eki, θ, G)
+
+Given forward map `G` and *unconstrained* parameters `θ`, return a 
+tuple `(Φ1, Φ2)` of terms in the EKI regularized objective function, where
+Φ = (1/2)*(Φ1 + Φ2). Φ1 measures output misfit `|| Γy^(-¹/₂) * (y .- G(θ)) ||²` and 
+Φ2 measures prior misfit `|| Γθ^(-¹/₂) * (θ .- μθ) ||²`, where `y` is the observation 
+map, `G(θ)` is the forward map, `Γy` is the observation noise covariance, `Γθ` is 
+the prior covariance, and `μθ` represents the prior means. Note that `Γ^(-1/2) = 
+inv(sqrt(Γ))`. 
+"""
+function eki_objective(eki, θ::AbstractVector, G::AbstractVector)
+    y = eki.mapped_observations
+    Γy = eki.noise_covariance
+
+    fp = eki.inverse_problem.free_parameters
+    unconstrained_priors = [unconstrained_prior(fp.priors[name]) for name in fp.names]
+    μθ = getproperty.(unconstrained_priors, :μ)
+    Γθ = diagm( getproperty.(unconstrained_priors, :σ).^2 )
+
+    # Φ1 = || Γy^(-¹/₂) * (y .- G) ||²
+    Φ1 = norm(inv(sqrt(Γy)) * (y .- G))^2
+    # Φ2 = || Γθ^(-¹/₂) * (θ .- μθ) ||² 
+    Φ2 = norm(inv(sqrt(Γθ)) * (θ .- μθ))^2
+    return (Φ1, Φ2)
```

Comment:
```suggestion
    # Φ₁ = || Γy^(-¹/₂) * (y - G) ||²
    Φ₁ = norm(inv(sqrt(Γy)) * (y .- G))^2
    # Φ₂ = || Γθ^(-¹/₂) * (θ - μθ) ||² 
    Φ₂ = norm(inv(sqrt(Γθ)) * (θ .- μθ))^2
    return (Φ₁, Φ₂)
```

aesthetically better?